### ISSUE TITLE
Sorting search results

### ISSUE THREAD

**Issue by ctubbsii** [sonatype-nexus-community/search-maven-org#44]
Thanks for creating an issue! Please fill out this form so we can be
sure to have all the information we need, and to minimize back and forth.

* **What are you trying to do?**
Sort search results (for example, sorting https://search.maven.org/search?q=g:org.apache.hadoop%20AND%20a:hadoop-client&core=gav by "Updated" column)

* **What feature or behavior is this required for?**
Finding information about a specific artifact or comparing similar artifacts. (For example, comparing similarly named artifacts with different groupIds, sorting by artifactId)

* **How could we solve this issue? (Not knowing is okay!)**
Could use something like https://datatables.net/ or provide a sortby directive in the search language.

* **Anything else?**
Nope. Keep up the good work!

**Comment by DarthHater** [comment]
This is a neat idea! We ultimately want to sort them by popularity at some point (as the default way we return them), as that seems like it would be very helpful for people. WDYT?

**Comment by ctubbsii** [comment]
Popularity might be useful, but sometimes what is popular isn't the best thing. For example, if multiple people fork servlet-api, and what becomes popular is (as example) Tomcat's package for servlet-api, but really the one that contains the canonical standard is the javax.servlet-api artifact, popularity can be misleading.

Having a default sort of popularity could be good, but I kind of like the current default (which appears to be "most recent"), because it tends to show me what I'm looking for, which is most often the answer to the question: is there a newer version of this?.

Regardless of default, having multiple sort options can support a wider variety of use cases.

**Comment by SirMaster** [comment]
@ctubbsii just as sorting on popularity, this would require some Back End changes to allow sorting on specific fields. 

This project is specifically related to the Front End of Search Maven Org, when we start looking into improving our Back End servers we will hopefully address this as well.

For now closing this, I did add a specific label that allows us to come back and review any changes we need to make once we updated out Back End.
